packages/compile-common/src/types.ts

```diff
@@ -64,12 +48,12 @@ export type CompiledContract = {
   source: string;
   sourceMap: string;
   deployedSourceMap: string;
-  legacyAST: object;
-  ast: object;
+  legacyAST?: object;
+  ast?: object;
   abi: Abi;
   metadata: string;
-  bytecode: Bytecode;
-  deployedBytecode: Bytecode;
+  bytecode?: Bytecode;
+  deployedBytecode?: Bytecode;
```

Comment:
This makes this PR a breaking change to compile-common, yes?

Reply:
Will this be breaking? It makes the type more lenient meaning that it shouldn't break anything, right?

Reply:
Yeah, should be considered non-breaking, IMO

Reply:
"Making the type more lenient" only applies if this optionality is an input type. Like, if we're talking about exporting some function`(options: { /* ... */ }) => any`, then adding a `?` to one of the `options` is non-breaking.

Here, it's different... to a package consuming the type, upgrading to the version that includes this change means handling a new case: previously, (e.g.) you would only need to handle `object`, but now you must handle both sides of `object | undefined`.

Reply:
Oh, very nice catch! I should've realized that this is a return type.